lib/pbench/server/api/resources/metadata_api.py

```diff
@@ -15,178 +15,167 @@ def __init__(self, config, logger, auth):
-                            {}]
-                        }
+                            "updated": "datetime string", }, ...]
                 }
         """
-        if not key:
+        if key is None:
             self.logger.warning("Metadata key not provided during metadata query")
-            abort(400, message="Missing metadata key in the query url")
+            abort(400, message="Missing metadata key in the get request uri")
 
         current_user = self.auth.token_auth.current_user()
         if current_user:
             current_user_id = current_user.id
         else:
             current_user_id = None
-        print(current_user_id)
         try:
-            # Fetch the metadata sessions
-            # If the key is favorite, we return only favorite sessions,
-            # else we return all the saved and favorite sessions
-            if key.upper() == "FAVORITE":
-                metadata_sessions = Metadata.query(
-                    user_id=current_user_id, key=MetadataKeys[key.upper()].value
-                )
-            else:
-                metadata_sessions = Metadata.query(user_id=current_user_id)
-            data = [session.get_json() for session in metadata_sessions]
+            # Query the metadata object with a given key
+            metadata_objects = Metadata.query(user_id=current_user_id, key=key.lower())
+            data = [
+                metadata.get_json(include=["id", "created", "updated", "value"])
+                for metadata in metadata_objects
+            ]
         except Exception:
             self.logger.exception(
                 "Exception occurred while querying the Metadata model"
             )
             abort(500, message="INTERNAL ERROR")
 
         response_object = {
-            "message": "success",
-            "data": {"sessions": data},
+            "data": data,
         }
         return make_response(jsonify(response_object), 200)
 
 
 class QueryMetadata(Resource):
     """
-    Abstracted pbench API for querying a single user metadata session
+    Abstracted pbench API for querying a single user metadata object
     """
 
     def __init__(self, config, logger):
         self.server_config = config
         self.logger = logger
 
-    def verify_metadata(self, session, action):
-        current_user_id = Auth.token_auth.current_user().id
-        metadata_user_id = session.user_id
-        if current_user_id != metadata_user_id:
+    def verify_metadata(self, metadata):
+        current_user = Auth.token_auth.current_user()
+        metadata_user_id = metadata.user_id
+        if current_user.id != metadata_user_id and not current_user.is_admin():
             self.logger.warning(
-                "Metadata {}: Logged in user_id {} is different than the one provided in the URI {}",
-                action,
-                current_user_id,
+                "Metadata user verification: Logged in user_id {} is different than the one provided in the URI {}",
```

Comment:
User isn't provided "in the URI". You're comparing the user_id in the metadata object against the token in the Authorization header.